client.go

```diff
@@ -0,0 +1,153 @@
+package support
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	neturl "net/url"
+)
+
+type HijackHttpOptions struct {
+	Method             string
+	Url                string
+	Success            chan struct{}
+	DockerTermProtocol bool
+	InputStream        io.Reader
+	ErrorStream        io.Writer
+	OutputStream       io.Writer
+	Data               interface{}
+	Header             http.Header
+	Log                Logger
+}
+
+// HijackHttpRequest performs an HTTP  request with given method, url and data and hijacks the request (after a successful connection) to stream
+// data from/to the given input, output and error streams.
+func HijackHttpRequest(options HijackHttpOptions) error {
+	if options.Log == nil {
+		// Make sure there is always a logger
+		options.Log = &logIgnore{}
+	}
+
+	req, err := createHijackHttpRequest(options)
+	if err != nil {
+		return err
+	}
+
+	// Parse URL for endpoint data
+	ep, err := neturl.Parse(options.Url)
+	if err != nil {
+		return err
+	}
+
+	protocol := ep.Scheme
+	address := ep.Path
+	if protocol != "unix" {
+		protocol = "tcp"
+		address = ep.Host
+	}
+
+	// Dial the server
+	var dial net.Conn
+	dial, err = net.Dial(protocol, address)
+	if err != nil {
+		return err
+	}
+
+	// Start initial HTTP connection
+	clientconn := httputil.NewClientConn(dial, nil)
```

Comment:
It's being done because it is lowlevel (which is what we need to hijack).
All hijacking code I found used this same pattern